shell/public/pwa-service-worker.js

```diff
@@ -0,0 +1,136 @@
+    "bug.svg",
+    "clipboard-m.svg",
+    "close-FFFFFF.svg",
+    "close-m.svg",
+    "close.svg",
+    "copy-m.svg",
+    "copy.svg",
+    "credit-m.svg",
+    "debug-m.svg",
+    "debug.svg",
+    "demo-logo-bg.svg",
+    "download-m.svg",
+    "download.svg",
+    "down-m.svg",
+    "down.svg",
+    "email-494949.svg",
+    "email-m.svg",
+    "email.svg",
+    "files-arrow.svg",
+    "github-color.svg",
+    "github-m.svg",
+    "github.svg",
+    "google-color.svg",
+    "google.svg",
+    "icons/icons-adec2a339de139470e74b4c487321525.svg",
+    "incognito.svg",
+    "install-714DAA.svg",
+    "install-896AC6.svg",
+    "keybase-m.svg",
+    "key-m.svg",
+    "key.svg",
+    "ldap-m.svg",
+    "ldap.svg",
+    "link-m.svg",
+    "link.svg",
+    "logos/uniregistry.svg",
+    "logos/draw.io-logo.svg",
+    "menu.svg",
+    "notification-m.svg",
+    "notification.svg",
+    "open-grain-m.svg",
+    "open-grain.svg",
+    "people-m.svg",
+    "people.svg",
+    "plus-6A237C.svg",
+    "plus-9E40B5.svg",
+    "pronoun-m.svg",
+    "question-727272.svg",
+    "question-a9a9a9.svg",
+    "referral-graphics/referral-1.svg",
+    "referral-graphics/referral-edge-rotated.svg",
+    "referral-graphics/referral-2.svg",
+    "referral-graphics/referral-3.svg",
+    "referral-graphics/referral-edge.svg",
+    "restart-m.svg",
+    "restart.svg",
+    "restore-5D5D5D.svg",
+    "restore-B7B7B7.svg",
+    "restore.svg",
+    "sandcat-aboutus.svg",
+    "sandstorm-for-work-lanyard.svg",
+    "sandstorm-gradient-logo.svg",
+    "search.svg",
+    "settings-m.svg",
+    "settings.svg",
+    "sfw-install.svg",
+    "share-m.svg",
+    "share.svg",
+    "source-m.svg",
+    "source.svg",
+    "standard-install.svg",
+    "trash-m.svg",
+    "trash.svg",
+    "troubleshoot-m.svg",
+    "troubleshoot.svg",
+    "twitter-m.svg",
+    "unlink-m.svg",
+    "upload-5D5D5D.svg",
+    "upload-B7B7B7.svg",
+    "upload.svg",
+    "up-m.svg",
+    "up.svg",
+    "web-m.svg",
+    "web.svg",
+    "devmode.png",
+    "icon32.png",
+    "icon.png",
+    "logos/jason.png",
+    "logos/kenton.png",
+    "logos/david.png",
+    "logos/miller.png",
+    "logos/luto.png",
+    "logos/seaborn.png",
+    "logos/audrey.png",
+    "logos/garply.png",
+    "logos/jasvir.png",
+    "logos/asheesh.png",
+    "logos/nena.png",
+    "logos/wagner.png",
+    "logos/swetland.png",
+    "logos/drew.png",
+    "logos/humanweb.png",
+    "logos/jade.png",
+    "sandstorm-gradient-logo.png",
+    "fonts/WOFF/OTF/SourceSansPro-Regular.otf.woff",
+    "fonts/WOFF/OTF/SourceSansPro-BoldIt.otf.woff",
+    "fonts/WOFF/OTF/SourceSansPro-Semibold.otf.woff",
+    "fonts/WOFF/OTF/SourceSansPro-It.otf.woff",
+    "fonts/WOFF/OTF/SourceSansPro-SemiboldIt.otf.woff",
+    "fonts/WOFF/OTF/SourceSansPro-Bold.otf.woff",
+    "icons/icons-adec2a339de139470e74b4c487321525.woff",
+  ]
+  event.waitUntil(caches.open('sandstorm-cache').then(function(cache) {
+    return cache.addAll(urlsToCache);
```

Comment:
Do we need to keep the list in-sync somehow?

What happens when the Sandstorm server is updated and the files change? Will the installation keep using the old versions?

Reply:
Re: sw updates, I remember getting an impression from the docs that it would auto-update the service worker. But yeah, looking back this is a little half-baked. I'm just going rip out the caching stuff entirely, and we can just to the bare minimum to get add to home screen working. Presumably browser caches will do an okay job on their own.